apps/token/src/routes/staking/associate/wallet-associate.tsx

```diff
@@ -35,19 +35,20 @@ export const WalletAssociate = ({
     appDispatch,
     appState: { walletBalance, allowance, walletAssociatedBalance, decimals },
   } = useAppState();
+  console.log(allowance.toString());
 
   const { token } = useContracts();
 
   const {
     state: approveState,
     perform: approve,
     dispatch: approveDispatch,
-  } = useTransaction(() =>
-    token.approve(
+  } = useTransaction(() => {
+    return token.approve(
       ethereumConfig.staking_bridge_contract.address,
-      Number.MAX_SAFE_INTEGER.toString()
-    )
-  );
+      removeDecimal(Number.MAX_SAFE_INTEGER.toString(), decimals).toString()
```

Comment:
question (non-blocking): I'm not sure why we decided to use Number.MAX_SAFE_INTEGER as the default approval amount. Seems a little weird and it might be nicer to just use a nice rounded number?

Reply:
I think it was back in the days where everything was 5 d.p. we can change it if we want, I have noticed a bug on the deposits screen for this. I'll change it to 1M.